Dunify project

psafont opened the pull request:
This was adapted from Marcello's and Tina's work on the vendored version in xen-api.
I have left out on purpose https://github.com/xapi-project/xen-api/commit/3f71785d6ca48474880ffcbadcd5fe2343574ba3 since this was not upstreamed at the time, so it seems it was not needed for the actual fix.

psafont committed:
maintenace: use dune to build the project

Signed-off-by: Pau Ruiz Safont <pau.safont@citrix.com>

psafont committed:
ci: update travis script

Lost coverage and doc upload

Signed-off-by: Pau Ruiz Safont <pau.safont@citrix.com>

psafont committed:
fix: redhat vendor string test

On newer distributions this string contains a dot

Signed-off-by: Pau Ruiz Safont <pau.safont@citrix.com>

psafont committed:
bindings: wrap the library inside Pci_bindings

Signed-off-by: Pau Ruiz Safont <pau.safont@citrix.com>

lindig commented on examples/dune:
```diff
@@ -0,0 +1,5 @@
+(executable
+  (name lspci)
+  (flags (:standard -safe-string))
```
Safe strings are standard in OCaml 4.08 at least - so not sure we need this explicitly.

lindig commented on pci.opam:
```diff
@@ -0,0 +1,35 @@
+opam-version: "2.0"
+synopsis: "Ctypes bindings to libpci for OCaml"
+maintainer: "simon.beaumont@citrix.com"
```
This address is no longer working. Maybe this repo should use `"xen-api@lists.xen.org"`

lindig commented on pci.opam:
```diff
@@ -0,0 +1,35 @@
+synopsis: "Ctypes bindings to libpci for OCaml"
+maintainer: "simon.beaumont@citrix.com"
+authors: "Si Beaumont"
+homepage: "https://github.com/simonjbeaumont/ocaml-pci"
```
Point to this repo?

lindig commented on stubgen/dune:
```diff
@@ -0,0 +1,23 @@
+(executables
+  (names ffi_stubgen ffi_types_stubgen)
+  (flags (:standard -safe-string))
```
Not needed any more?

lindig commented on .travis.yml:
```diff
@@ -1,15 +1,32 @@
+  - wget https://raw.githubusercontent.com/ocaml/ocaml-ci-scripts/master/.travis-opam.sh
+script: bash -ex .travis-opam.sh
+jobs:
+  include:
```
Is this the list of builds run on Travis? It seems a bit excessive. Maybe limit this to 4.07 to 4.10?

psafont replied:
All Ocaml versions are supported upstream, starting from 4.02. There is even one fix you backported so the code would work in versions prior to 4.05.

lindig commented on bindings/dune:
```diff
@@ -0,0 +1,6 @@
+(library
+  (name pci_bindings)
+  (public_name pci.bindings)
+  (flags (:standard -safe-string))
```
Not needed for recent OCaml versions

psafont committed:
opam: change contact addresses

psafont committed:
CA-293399: Use original fix, raise min ocaml to 4.08

The original fix for this bug has been tested for years, while the one
used for upstream doesn't seem to have had that much usage.
https://github.com/xapi-project/xen-api/commit/bc053bbe0b930c58f0da403a5e8ac6bcdeb7a026

Since the original fix needs ocaml 4.05, I used the opportunity to raise
the minimum version to 4.08.

Signed-off-by: Pau Ruiz Safont <pau.safont@citrix.com>

lindig commented:
Are these left to do after the merge?
* Switch xapi to use this and remove the vendored package?
* Add this to xs-opam?

psafont commented:
Yep, those are to do.

I checked if the xapi spec file installed any binary, but this isn't happening. It's only used as a library as far as I can see.

lindig commented:
I'd be ready to merge this but maybe have a second pair of eyes take a look.

psafont commented:
I'll tag these changes as `v1.0.2` so it can be added to `xs-opam`.
